src/WebJobs.Script/Host/HostFunctionMetadataProvider.cs

```diff
@@ -179,10 +183,21 @@ internal static string ParseLanguage(string scriptFilePath, IEnumerable<RpcWorke
 
             // determine the script type based on the primary script file extension
             string extension = Path.GetExtension(scriptFilePath).ToLowerInvariant().TrimStart('.');
-            var workerConfig = workerConfigs.FirstOrDefault(config => config.Description.Extensions.Contains("." + extension));
-            if (workerConfig != null)
+
+            // In the special case of a null FUNCTIONS_WORKER_RUNTIME and an extension of "dll",
```

Comment:
There's an assumption made by this method that the value passed in the parameter is coming from reading `FUNCTIONS_WORKER_RUNTIME`, while the method uses what is passed in, which could eventually come from different sources. Would be good to update the comment to avoid confusion here in the future.